azure/src/main/java/org/apache/iceberg/azure/adlsv2/ADLSv2Location.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.azure.adlsv2;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/** This class represents a fully qualified location in Azure expressed as a URI. */
+class ADLSv2Location {
+  private static final Pattern URI_PATTERN =
+      Pattern.compile("^abfss?://(.+?)@(.+?\\.dfs\\.core\\.windows\\.net)(/.*)?$");
```

Comment:
Two things:
- At Azure, you can alias your custom domain name: https://learn.microsoft.com/en-us/azure/storage/blobs/storage-custom-domain-name?tabs=azure-portal
- It is common to have a [privatelink](https://learn.microsoft.com/en-us/azure/storage/common/storage-private-endpoints). By default traffic in Azure goes over the public internet, with a private link you create a separate DNS entry with a local address, keeping all the traffic on your private network.

Reply:
Thanks for pointing that out, I removed the domain name validation so it takes the host name as-is.